### PR TITLE
Feat/landingpage

### PR DESCRIPTION
**Pull request by Quekyy**
Ready for Review, entirely UI. 
Created Landing Page similar to Google Drive's Menu.
Details Modal to be coded another time.

### CODE REVIEW

File: apps/backend/src/routes/home/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+<script>
+    // Your Svelte script here
+</script>
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Home</title>
+    <script src="https://kit.fontawesome.com/c850f0d795.js" crossorigin="anonymous"></script>
+</head>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:11]: Not allowed in svelte

File: apps/backend/src/routes/home/+page.svelte
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:1]: Ensure that your indents are SPACES and indents of 2 only.

File: apps/backend/src/routes/home/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,133 @@
+<!-- JavaScript code -->
+<script>
+    // Your Svelte script here
+</script>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:4]: Use `lang='ts'`. Do not use normal Javascript

File: apps/backend/src/routes/home/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Home</title>
+    <script src="https://kit.fontawesome.com/c850f0d795.js" crossorigin="anonymous"></script>
+</head>
+
+<!-- Landing Page - Card -->
+<div class="flex items-center justify-center h-screen">
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:14]: Specify explicitly if its `flex-row` or `flex-col`

File: apps/backend/src/routes/home/+page.svelte [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    <!-- Landing Page - Content -->
+    <div class="w-11/12 h-screen p-10 border border-secondary rounded-md shadow-lg">
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:17]: Do not use fractions such as `w-11/12`

File: apps/backend/src/routes/home/+page.svelte [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            <div tabindex="0" role="button" class="btn m-1 text-2xl font-semibold normal-case px-6 rounded-full border-none bg-transparent">My Files
+                <span>
+                    <i class="fa-solid fa-caret-down "></i>
+                </span>
+            </div>
+            <ul class="dropdown-content z-[1] menu p-2 shadow-2xl bg-base-100 rounded-box w-52 border">
+                <li><a href="/"><i class="fa-solid fa-folder"></i> New Folder</a></li>
+                <hr>
+                <li><a href="/"><i class="fa-solid fa-file-arrow-up"></i> File Upload</a></li>
+                <li><a href="/"><i class="fa-solid fa-file-arrow-up"></i> Folder Upload</a></li>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:30]: Do not inline all the code. Separate all the code into different lines

File: apps/backend/src/routes/home/+page.svelte [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                    <span>
+                        <i class="fa-solid fa-caret-down "></i>
+                    </span>
+                </div>
+                <ul class="dropdown-content z-[1] menu p-2 shadow-2xl bg-base-100 rounded-box w-52 border">
+                    <li><a href="/"><i class="fa-solid fa-folder"></i>Open Class</a></li>
+                    <hr>
+                    <li><a href="/"><i class="fa-solid fa-folder"></i>Confidential Class</a></li>
+                    <hr>
+                    <li><a href="/"><i class="fa-solid fa-folder"></i>Secret Class</a></li>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:49]: Do not inline all the code. Separate all the code into different lines

File: apps/backend/src/routes/home/+page.svelte [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+                    <span>
+                        <i class="fa-solid fa-caret-down "></i>
+                    </span>
+                </div>
+                <ul class="dropdown-content z-[1] menu p-2 shadow-2xl bg-base-100 rounded-box w-52 border">
+                    <li><a href="/"><i class="fa-solid fa-folder"></i>Today</a></li>
+                    <li><a href="/"><i class="fa-solid fa-folder"></i>Last 7 Days</a></li>
+                    <li><a href="/"><i class="fa-solid fa-folder"></i>Last 30 Days</a></li>
+                    <li><a href="/"><i class="fa-solid fa-folder"></i>This Year (2023)</a></li>
+                    <li><a href="/"><i class="fa-solid fa-folder"></i>Last Year (2022)</a></li>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:65]: Do not inline all the code. Separate all the code into different lines

File: apps/backend/src/routes/home/+page.svelte (hunk 1 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:11]: Removed <head> tag and its contents. Fixed the font awesome by using svg

File: apps/backend/src/routes/home/+page.svelte (hunk 4 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:17]: fixed `with w-full max-w-7xl` and removed `border-secondary`

File: apps/backend/src/routes/home/+page.svelte (hunk 6 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:49]: Lines of codes separated, Class removed

Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:49]: Separated lines of codes for all pages here as well

File: apps/backend/src/routes/home/+page.svelte (hunk 7 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:65]: Fixed it using `const currentYear = (new Date().getFullYear() as unknown) as string;`

All codes not inlined

File: apps/backend/src/routes/login/+page.svelte
[the remark this answers is not in the document]
Comment (Quekyy) [reply apps/backend/src/routes/login/+page.svelte:1]: Done in previous PR

File: apps/backend/src/routes/signup/+page.svelte
[the remark this answers is not in the document]
Comment (Quekyy) [reply apps/backend/src/routes/signup/+page.svelte:1]: Done in previous PR

File: apps/backend/src/routes/signup/onboard/+page.svelte
[the remark this answers is not in the document]
Comment (Quekyy) [reply apps/backend/src/routes/signup/onboard/+page.svelte:1]: Done in previous PR

File: apps/backend/src/routes/home/+page.svelte
Comment (taronaeo) [reply apps/backend/src/routes/home/+page.svelte:1]: image

Spacing of 2 is required

File: apps/backend/src/routes/home/+page.svelte [hunk 8]
```diff
@@ -0,0 +1,240 @@
+<!-- JavaScript code -->
+<script lang='ts'>
+  //Script to obtain current year
+  const currentYear = (new Date().getFullYear() as unknown) as string;
+
+  //Script to obtian Previous year
+  const previousYear = ((new Date().getFullYear() - 1) as unknown) as string;
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:7]: DO NOT DO THIS, MASSIVE ISSUE! `.getFullYear()` already returns a `number` type. All you need is to do `.toString()` to convert it to a `string`.

Comment (taronaeo) [reply apps/backend/src/routes/home/+page.svelte:7]: Also, every time you do `new Date()`, it initialises a new `Date` object in memory, which will build up over time if you keep initialising new ones.

Save the `new Date()` object to a const variable and use it from there.

File: apps/backend/src/routes/home/+page.svelte [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+    <!-- Landing Page - Content -->
+    <div class=" w-full max-w-7xl h-screen p-10 rounded-md">
+        
+        <!-- Landing Page - Header -->
+        <div class="dropdown dropdown-bottom">
+            <div tabindex="0" role="button" class="btn m-1 text-2xl font-semibold normal-case px-6 rounded-full border-none bg-transparent">My Files
+                <span>
+                    <svg xmlns="http://www.w3.org/2000/svg" height="20" width="14" viewBox="0 0 320 512" fill="#fff">
+                        <!--!Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2023 Fonticons, Inc.--><path d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"/>
+                    </svg>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:23]: Save this `svg` into a file of it's own and use it dynamically via imports.

File: apps/backend/src/routes/home/+page.svelte [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+                </span>
+            </div>
+            <ul class="dropdown-content z-[1] menu p-2 shadow-2xl bg-base-100 rounded-box w-52 border">
+                <li>
+                    <a href="/">
+                        <svg xmlns="http://www.w3.org/2000/svg" height="16" width="12" viewBox="0 0 384 512" fill="#fff">
+                            <!--!Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2023 Fonticons, Inc.--><path d="M0 64C0 28.7 28.7 0 64 0H224V128c0 17.7 14.3 32 32 32H384V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V64zm384 64H256V0L384 128z"/>
+                        </svg>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:31]: Save this svg into a file of it's own and use it dynamically via imports.

File: apps/backend/src/routes/home/+page.svelte [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+                         New Folder
+                    </a>
+                </li>
+                <hr>
+                <li>
+                    <a href="/">
+                        <svg xmlns="http://www.w3.org/2000/svg" height="16" width="12" viewBox="0 0 384 512" fill="#fff">
+                            <!--!Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2023 Fonticons, Inc.--><path d="M0 64C0 28.7 28.7 0 64 0H224V128c0 17.7 14.3 32 32 32H384V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V64zm384 64H256V0L384 128z"/>
+                        </svg>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:40]: Save this svg into a file of it's own and use it dynamically via imports.

File: apps/backend/src/routes/home/+page.svelte [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+                         File Upload
+                    </a>
+                </li>
+                <li>
+                    <a href="/">
+                        <svg xmlns="http://www.w3.org/2000/svg" height="16" width="12" viewBox="0 0 384 512" fill="#fff">
+                            <!--!Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2023 Fonticons, Inc.--><path d="M64 0C28.7 0 0 28.7 0 64V448c0 35.3 28.7 64 64 64H320c35.3 0 64-28.7 64-64V160H256c-17.7 0-32-14.3-32-32V0H64zM256 0V128H384L256 0zM216 408c0 13.3-10.7 24-24 24s-24-10.7-24-24V305.9l-31 31c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l72-72c9.4-9.4 24.6-9.4 33.9 0l72 72c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-31-31V408z"/>
+                        </svg>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:48]: Save this svg into a file of it's own and use it dynamically via imports.

File: apps/backend/src/routes/home/+page.svelte [hunk 13]
```diff
@@ -0,0 +1,240 @@
...
+        <!-- Dropdown buttons for User -->
+        <div>
+
+            <!-- Type Dropdown -->
+            <div class="dropdown dropdown-bottom">
+                <div tabindex="0" role="button" class="btn m-1 font-semibold normal-case px-6 border-white hover:border-white">Classification
+                    <span>
+                        <svg xmlns="http://www.w3.org/2000/svg" height="20" width="14" viewBox="0 0 320 512" fill="#fff">
+                            <!--!Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2023 Fonticons, Inc.--><path d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"/>
+                        </svg>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:64]: Save this svg into a file of it's own and use it dynamically via imports.

File: apps/backend/src/routes/home/+page.svelte [hunk 14]
```diff
@@ -0,0 +1,240 @@
...
+                </ul>
+            </div>
+
+            <!-- Type Dropdown -->
+            <div class="dropdown dropdown-bottom">
+                <div tabindex="0" role="button" class="btn m-1 font-semibold normal-case px-6 border-white hover:border-white">Modified
+                    <span>
+                        <svg xmlns="http://www.w3.org/2000/svg" height="20" width="14" viewBox="0 0 320 512" fill="#fff">
+                            <!--!Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2023 Fonticons, Inc.--><path d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"/>
+                        </svg>
```
Comment (taronaeo) [inline apps/backend/src/routes/home/+page.svelte:94]: Save this svg into a file of it's own and use it dynamically via imports.

File: apps/backend/src/routes/login/+page.svelte
Comment (taronaeo) [inline apps/backend/src/routes/login/+page.svelte:1]: image

Spacing of 2 is required

File: apps/backend/src/routes/login/+page.svelte [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+            <!-- Form Content - Email Input -->
+            <div>
+                <label class="form-control w-full max-w-md">
+                    <div class="label">
+                        <span class="label-text">
+                          Email
+                        </span>
+                    </div>
+                    <input type="text" placeholder="Name@example.com" class="input input-bordered w-full max-w-md 
+                    focus:ring-primary focus:border-primary duration-300 focus:outline-none" />
```
Comment (taronaeo) [inline apps/backend/src/routes/login/+page.svelte:31]: Missing `name` attribute

Comment (taronaeo) [reply apps/backend/src/routes/login/+page.svelte:31]: Also, space out your classes into multiple lines

File: apps/backend/src/routes/login/+page.svelte [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+            <!-- Form Content - Password Input -->
+            <div>
+                <label class="form-control w-full max-w-md">
+                    <div class="label">
+                        <span class="label-text">
+                          Password
+                        </span>
+                    </div>
+                    <input type="text" placeholder="••••••••" class="input input-bordered w-full max-w-md 
+                    focus:ring-primary focus:border-primary duration-300 focus:outline-none" />
```
Comment (taronaeo) [inline apps/backend/src/routes/login/+page.svelte:44]: Missing `name` attribute

File: apps/backend/src/routes/login/+page.svelte [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+                        viewBox="0 0 24 24"
+                    >
+                        <circle class="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" stroke-width="4"
+                        ></circle>
+                        <path
+                            class="opacity-75"
+                            fill="currentColor"
+                            d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
+                        ></path>
+                    </svg>
```
Comment (taronaeo) [inline apps/backend/src/routes/login/+page.svelte:74]: Save this `svg` into a file of it's own and use it dynamically via imports.

**Review by taronaeo — CHANGES_REQUESTED** [review]
Wait for PR #19 to be resolved and merged before requesting another review

**Comment by taronaeo** [comment]
Please resolve all conflicts as well before requesting another review

File: apps/backend/src/routes/home/+page.svelte
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:1]: image

Updated

File: apps/backend/src/routes/home/+page.svelte (hunk 8 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:7]: Changed accordingly:

Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:7]: Changed Accordingly:

```
// Script to create a Current Date object
const currentDate = new Date();

//Script to obtain current year
const currentYear = currentDate.getFullYear().toString();

//Script to obtian Previous year
const previousYear = (currentDate.getFullYear() - 1).toString();
```

File: apps/backend/src/routes/login/+page.svelte
Comment (Quekyy) [reply apps/backend/src/routes/login/+page.svelte:1]: image

Spacing fixed

File: apps/backend/src/routes/home/+page.svelte (hunk 9 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:23]: Didn't use imports, instead simply added svg files into static and used an `` tag. Let me know if this is okay.

image

File: apps/backend/src/routes/home/+page.svelte (hunk 10 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:31]: Done the same as above, added a new file called fileicon.svg in static and used an `` tag

File: apps/backend/src/routes/home/+page.svelte (hunk 11 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:40]: Same as all the other svg fixes, added a new file called fileupload.svg and did the same

File: apps/backend/src/routes/home/+page.svelte (hunk 12 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:48]: Did the same for all other svg fixes

File: apps/backend/src/routes/home/+page.svelte (hunk 13 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:64]: Did the same for all other svg fixes

File: apps/backend/src/routes/home/+page.svelte (hunk 14 above)
Comment (Quekyy) [reply apps/backend/src/routes/home/+page.svelte:94]: Did the same for all other svg fixes

File: apps/backend/src/routes/login/+page.svelte (hunk 15 above)
Comment (Quekyy) [reply apps/backend/src/routes/login/+page.svelte]: Spaced out classes. Added Name attribute from the #19 fix.

File: apps/backend/src/routes/login/+page.svelte (hunk 16 above)
Comment (Quekyy) [reply apps/backend/src/routes/login/+page.svelte]: Same as above

File: apps/backend/src/routes/login/+page.svelte (hunk 17 above)
Comment (Quekyy) [reply apps/backend/src/routes/login/+page.svelte]: This has a different fix, also from #19. Use daisyUI button to fix:

image

**Comment by taronaeo** [comment]
Closing as requested by @Quekyy

**Pull request closed without merging**